gax/src/test/java/com/google/api/gax/rpc/OperationCallableImplTest.java

```diff
@@ -0,0 +1,983 @@
+import java.awt.Color;
+import java.io.IOException;
+import java.util.Currency;
+import java.util.concurrent.CancellationException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.threeten.bp.Duration;
+
+@RunWith(JUnit4.class)
+public class OperationCallableImplTest {
+  private CallableFactory callableFactory =
+      CallableFactory.create(FakeTransportDescriptor.create());
+
+  private static final RetrySettings FAST_RETRY_SETTINGS =
+      RetrySettings.newBuilder()
+          .setInitialRetryDelay(Duration.ofMillis(1L))
+          .setRetryDelayMultiplier(1)
+          .setMaxRetryDelay(Duration.ofMillis(1L))
+          .setInitialRpcTimeout(Duration.ofMillis(1L))
+          .setMaxAttempts(0)
+          .setJittered(false)
+          .setRpcTimeoutMultiplier(1)
+          .setMaxRpcTimeout(Duration.ofMillis(1L))
+          .setTotalTimeout(Duration.ofMillis(5L))
+          .build();
+
+  private FakeChannel initialChannel;
+  private FakeChannel pollChannel;
+  private TransportChannel pollTransportChannel;
+  private RecordingScheduler executor;
+  private ClientContext initialContext;
+  private OperationCallSettings<Integer, Color, Currency> callSettings;
+
+  private FakeApiClock clock;
+  private OperationTimedPollAlgorithm pollingAlgorithm;
+
+  @Before
+  public void setUp() throws IOException {
+    initialChannel = mock(FakeChannel.class);
+    pollChannel = mock(FakeChannel.class);
+    pollTransportChannel = mock(TransportChannel.class);
+    TransportChannelProvider operationsChannelProvider = mock(TransportChannelProvider.class);
+    when(operationsChannelProvider.getTransportChannel()).thenReturn(pollTransportChannel);
+
+    clock = new FakeApiClock(0L);
+    executor = RecordingScheduler.create(clock);
+    pollingAlgorithm = new OperationTimedPollAlgorithm(FAST_RETRY_SETTINGS, clock);
+
+    SimpleCallSettings<Integer, OperationSnapshot> initialCallSettings =
+        SimpleCallSettings.<Integer, OperationSnapshot>newBuilder()
+            .setRetrySettings(FAST_RETRY_SETTINGS.toBuilder().setMaxAttempts(1).build())
+            .build();
+
+    callSettings =
+        OperationCallSettings.<Integer, Color, Currency>newBuilder()
+            .setInitialCallSettings(initialCallSettings)
+            .setResponseTransformer(new ResponseTransformer())
+            .setMetadataTransformer(new MetadataTransformer())
+            .setPollingAlgorithm(pollingAlgorithm)
+            .build();
+
+    initialContext = getClientContext(initialChannel, executor);
+  }
+
+  private static class ResponseTransformer implements ApiFunction<OperationSnapshot, Color> {
```

Comment:
This is suspicious. Those transformers are essential part of what was tested by this test. Now, they are the fake ones, as result it looks like the test is not testing actual operations anymore (it is impoortant, because the longrunning operations logic relies on those transformers to be correct).

Reply:
This is a consequence of the operation logic moving to the core level; the transformers are transport-specific and can't be moved up. We'll need to have tests at the grpc level for the transformations (still TO DO for the PR).

Reply:
Are you going to add the tests as part of this PR?